### PR TITLE
Addresses changes made to security demo config install tool

### PR DESCRIPTION
**Pull request by DarshitChanpura**
### Description
There were recent changes to security demo configuration setup which now requires a custom admin Password to be setup. This PR updates the documentation and related code change.
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/search-processor/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by msfroh** [comment]
Thanks @DarshitChanpura! 

Before we merge this, I'm going to make some changes to update the quickstart scripts to actually use newer versions of OpenSearch. (Currently, the Kendra one still downloads OpenSearch 2.7 and the Personalize one downloads 2.9.)

**Review by derek-ho — APPROVED** [review]
LGTM

File: helpers/personalized_search_ranking_quickstart.sh
Comment (derek-ho) [inline helpers/personalized_search_ranking_quickstart.sh]: This doesn't work comparing 2.7 vs 2.12 - I suggest:
```
IFS='.' read -ra version_array <<< "$VERSION"



if [ -z "$CREDENTIAL" ]
then
  # Starting in 2.12.0, security demo configuration script requires an initial admin password
  if (( ${version_array[0]} > 2 || (${version_array[0]} == 2 && ${version_array[1]} >= 12) )); then
      CREDENTIAL="admin:myStrongPassword123!"
  else
      CREDENTIAL="admin:admin"
  fi
fi
```

Comment (derek-ho) [reply helpers/personalized_search_ranking_quickstart.sh]: Which is the logic I used in other repos

**Comment by DarshitChanpura** [comment]
@msfroh @macohen @macohen @noCharger @mingshl @sejli Can one of the maintainers take a look at the failing CI and help bring this to closure?

File: helpers/personalized_search_ranking_quickstart.sh
Comment (msfroh) [inline helpers/personalized_search_ranking_quickstart.sh]: Rather than hard-coding, we should probably take the admin password as a command-line arg.

@DarshitChanpura -- did earlier versions support the `OPENSEARCH_INITIAL_ADMIN_PASSWORD` env var? Or does that change come in 2.12? 

I'm wondering if it makes sense to change the scripts to require the initial admin password for all versions, where it will be required for 2.12 and a good idea for earlier versions.

Comment (DarshitChanpura) [reply helpers/personalized_search_ranking_quickstart.sh]: > Rather than hard-coding, we should probably take the admin password as a command-line arg.

Starting 2.12 a custom strong password is required. 

> @DarshitChanpura -- did earlier versions support the OPENSEARCH_INITIAL_ADMIN_PASSWORD env var? Or does that change come in 2.12?

This change is introduced in 2.12 and will be present for all versions 2.12 and above

> I'm wondering if it makes sense to change the scripts to require the initial admin password for all versions, where it will be required for 2.12 and a good idea for earlier versions.

That might be your design decision as a maintainer. Since the admin password is required only for 2.12 and above, IMO we can keep a version check in place and require it only for >= 2.12

Comment (msfroh) [reply helpers/personalized_search_ranking_quickstart.sh]: This doesn't answer my question. Allow me to rephrase:

Is it **POSSIBLE** to set the admin password in `docker-compose.yml` before 2.12?

If so, I will change the script to always require a custom password, for all versions, since it's a good idea.

Comment (DarshitChanpura) [reply helpers/personalized_search_ranking_quickstart.sh]: > Is it POSSIBLE to set the admin password in docker-compose.yml before 2.12?

You can set the env variable `OPENSEARCH_INITIAL_ADMIN_PASSWORD` in the docker-compose.yml. This value will only be picked up for 2.12 and above

You, can change the script to have a default strong password, and then based on the version it will be picked.

Does that answer your query?

Comment (msfroh) [reply helpers/personalized_search_ranking_quickstart.sh]: Yes, thank you.

In that case, it does make more sense to accept and enforce it for version 2.12 and higher only, because requiring the parameter on earlier versions and then ignoring it (leaving the default password as `admin`) would be a terrible user experience. 

In fact, we should reject the parameter (or at least output a warning that it's being ignored) if the version is pre-2.12.

I'll make those changes to the script.

Comment (msfroh) [reply helpers/personalized_search_ranking_quickstart.sh]: Made those changes and pushed the commit to this PR. 

@DarshitChanpura, please let me know what you think. Thanks!

**Review by DarshitChanpura — COMMENTED** [review]
LGTM!

**Review by msfroh — APPROVED** [review]

**Review by noCharger — APPROVED** [review]
LGTM

**Comment by msfroh** [comment]
> The backport to 2.12 failed:

That makes sense, since we haven't cut a 2.12 branch yet -- it will get cut from 2.x.

I'm just going to remove the `backport-2.12` label.

**Pull request merged**
